### ISSUE TITLE
RDS Drift pollution

### ISSUE THREAD

**Issue by Staggerlee011** [cloudskiff/driftctl#327]
Hi when running a `driftctl` against a `rds` instance statefile we got various drift notifications that i would take as noise / not needed.  

The RDS instances are created via the module: `https://github.com/terraform-aws-modules/terraform-aws-rds`

driftctl output:

```
Found drifted resources:
  - confluence (aws_db_instance):
    ~ AllowMajorVersionUpgrade: false => <nil>
    ~ ApplyImmediately: false => <nil> (computed)
    ~ LatestRestorableTime: <nil> => "2021-03-10T09:20:40Z" (computed)
    ~ SnapshotIdentifier: "" => <nil>
  - confluence-20210107152452716900000003 (aws_db_subnet_group):
    ~ NamePrefix: "confluence-" => <nil> (computed)
```

- aws_db_subnet_group: is not defined on build, i think AWS just uses it to link the desired subnets together, so should not considered drift
- `"" => <nil>` and `false => <nil>` should not be considered drift
- `LatestRestorableTime` will be ever changing so should not considered drift

statefile for rds related resources:

```
{
      "module": "module.db.module.db_instance",
      "mode": "data",
      "type": "aws_iam_policy_document",
      "name": "enhanced_monitoring",
      "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "id": "1813475199",
            "json": "{\n  \"Version\": \"2012-10-17\",\n  \"Statement\": [\n    {\n      \"Sid\": \"\",\n      \"Effect\": \"Allow\",\n      \"Action\": \"sts:AssumeRole\",\n      \"Principal\": {\n        \"Service\": \"monitoring.rds.amazonaws.com\"\n      }\n    }\n  ]\n}",
            "override_json": null,
            "policy_id": null,
            "source_json": null,
            "statement": [
              {
                "actions": [
                  "sts:AssumeRole"
                ],
                "condition": [],
                "effect": "Allow",
                "not_actions": [],
                "not_principals": [],
                "not_resources": [],
                "principals": [
                  {
                    "identifiers": [
                      "monitoring.rds.amazonaws.com"
                    ],
                    "type": "Service"
                  }
                ],
                "resources": [],
                "sid": ""
              }
            ],
            "version": "2012-10-17"
          },
          "sensitive_attributes": []
        }
      ]
    },
    {
      "module": "module.db.module.db_instance",
      "mode": "managed",
      "type": "aws_db_instance",
      "name": "this",
      "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
      "instances": [
        {
          "index_key": 0,
          "schema_version": 1,
          "attributes": {
            "address": "confluence.xxx.eu-west-2.rds.amazonaws.com",
            "allocated_storage": 5,
            "allow_major_version_upgrade": false,
            "apply_immediately": false,
            "arn": "arn:aws:rds:eu-west-2:xxx:db:confluence",
            "auto_minor_version_upgrade": true,
            "availability_zone": "eu-west-2a",
            "backup_retention_period": 1,
            "backup_window": "03:00-06:00",
            "ca_cert_identifier": "rds-ca-2019",
            "character_set_name": null,
            "copy_tags_to_snapshot": false,
            "db_subnet_group_name": "confluence-20210107152452716900000003",
            "delete_automated_backups": true,
            "deletion_protection": false,
            "domain": "",
            "domain_iam_role_name": "",
            "enabled_cloudwatch_logs_exports": [
              "postgresql",
              "upgrade"
            ],
            "endpoint": "xxx.com:54321",
            "engine": "postgres",
            "engine_version": "12.5",
            "final_snapshot_identifier": null,
            "hosted_zone_id": "Z1TTGA775OQIYO",
            "iam_database_authentication_enabled": true,
            "id": "confluence",
            "identifier": "confluence",
            "identifier_prefix": null,
            "instance_class": "db.t2.small",
            "iops": 0,
            "kms_key_id": "arn:aws:kms:eu-west-2:xxx:key/e531df1d-c46e-4b96-a948-96c41e37608b",
            "license_model": "postgresql-license",
            "maintenance_window": "mon:00:00-mon:03:00",
            "max_allocated_storage": 0,
            "monitoring_interval": 60,
            "monitoring_role_arn": "arn:aws:iam::5xxx:role/engineering-monitoring-role",
            "multi_az": false,
            "name": "confluence",
            "option_group_name": "confluence-20210107151803555200000002",
            "parameter_group_name": "confluence-20210107151803555000000001",
            "password": "xxxxxxxxxxxx",
            "performance_insights_enabled": false,
            "performance_insights_kms_key_id": "",
            "performance_insights_retention_period": 0,
            "port": 54321,
            "publicly_accessible": false,
            "replicas": [],
            "replicate_source_db": "",
            "resource_id": "db-GHALF2SDV3ES5ZIHWIOV2QH4TM",
            "s3_import": [],
            "security_group_names": [],
            "skip_final_snapshot": true,
            "snapshot_identifier": "",
            "status": "available",
            "storage_encrypted": true,
            "storage_type": "gp2",
            "tags": {
              "Application": "RDS",
              "Automated-stop-start": "true",
              "Environment": "psn-dev-eks-vpc",
              "Name": "confluence",
              "Terraform": "true"
            },
            "timeouts": {
              "create": "40m",
              "delete": "40m",
              "update": "80m"
            },
            "timezone": "",
            "username": "postgres",
            "vpc_security_group_ids": [
              "sg-0517eeef3af9e5163"
            ]
          },
          "sensitive_attributes": [],
          "private": "eyJlMmJmYjczMC1lY2FhLTExZTYtOGY4OC0zNDM2M2JjN2M0YzAiOnsiY3JlYXRlIjoyNDAwMDAwMDAwMDAwLCJkZWxldGUiOjI0MDAwMDAwMDAwMDAsInVwZGF0ZSI6NDgwMDAwMDAwMDAwMH0sInNjaGVtYV92ZXJzaW9uIjoiMSJ9",
          "dependencies": [
            "data.aws_subnet_ids.subnets",
            "data.aws_vpc.vpc",
            "module.db.module.db_instance.aws_iam_role.enhanced_monitoring",
            "module.db.module.db_instance.data.aws_iam_policy_document.enhanced_monitoring",
            "module.db.module.db_option_group.aws_db_option_group.this",
            "module.db.module.db_parameter_group.aws_db_parameter_group.this",
            "module.db.module.db_parameter_group.aws_db_parameter_group.this_no_prefix",
            "module.db.module.db_subnet_group.aws_db_subnet_group.this",
            "module.rds-sg.aws_security_group.this",
            "module.rds-sg.aws_security_group.this_name_prefix",
            "random_password.pwd"
          ]
        }
      ]
    },
    {
      "module": "module.db.module.db_instance",
      "mode": "managed",
      "type": "aws_iam_role",
      "name": "enhanced_monitoring",
      "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
      "instances": [
        {
          "index_key": 0,
          "schema_version": 0,
          "attributes": {
            "arn": "arn:aws:iam::xxx:role/engineering-monitoring-role",
            "assume_role_policy": "{\"Version\":\"2012-10-17\",\"Statement\":[{\"Sid\":\"\",\"Effect\":\"Allow\",\"Principal\":{\"Service\":\"monitoring.rds.amazonaws.com\"},\"Action\":\"sts:AssumeRole\"}]}",
            "create_date": "2020-12-01T15:06:08Z",
            "description": "",
            "force_detach_policies": false,
            "id": "engineering-monitoring-role",
            "max_session_duration": 3600,
            "name": "engineering-monitoring-role",
            "name_prefix": null,
            "path": "/",
            "permissions_boundary": null,
            "tags": {
              "Application": "RDS",
              "Automated-stop-start": "true",
              "Environment": "xxx-vpc",
              "Name": "engineering-monitoring-role",
              "Terraform": "true"
            },
            "unique_id": "AROAYPP77LT3PT25GZKOA"
          },
          "sensitive_attributes": [],
          "private": "bnVsbA==",
          "dependencies": [
            "data.aws_vpc.vpc",
            "module.db.module.db_instance.data.aws_iam_policy_document.enhanced_monitoring"
          ]
        }
      ]
    },
    {
      "module": "module.db.module.db_instance",
      "mode": "managed",
      "type": "aws_iam_role_policy_attachment",
      "name": "enhanced_monitoring",
      "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
      "instances": [
        {
          "index_key": 0,
          "schema_version": 0,
          "attributes": {
            "id": "engineering-monitoring-role-20201201150607855000000004",
            "policy_arn": "arn:aws:iam::aws:policy/service-role/AmazonRDSEnhancedMonitoringRole",
            "role": "engineering-monitoring-role"
          },
          "sensitive_attributes": [],
          "private": "bnVsbA==",
          "dependencies": [
            "data.aws_vpc.vpc",
            "module.db.module.db_instance.aws_iam_role.enhanced_monitoring",
            "module.db.module.db_instance.data.aws_iam_policy_document.enhanced_monitoring"
          ]
        }
      ]
    },
    {
      "module": "module.db.module.db_option_group",
      "mode": "managed",
      "type": "aws_db_option_group",
      "name": "this",
      "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
      "instances": [
        {
          "index_key": 0,
          "schema_version": 0,
          "attributes": {
            "arn": "arn:aws:rds:eu-west-2:xxx:og:confluence-20210107151803555200000002",
            "engine_name": "postgres",
            "id": "confluence-20210107151803555200000002",
            "major_engine_version": "12",
            "name": "confluence-20210107151803555200000002",
            "name_prefix": "confluence-",
            "option": [],
            "option_group_description": "Option group for confluence",
            "tags": {
              "Application": "RDS",
              "Automated-stop-start": "true",
              "Environment": "xxx-vpc",
              "Name": "confluence",
              "Terraform": "true"
            },
            "timeouts": {
              "delete": "15m"
            }
          },
          "sensitive_attributes": [],
          "private": "eyJlMmJmYjczMC1lY2FhLTExZTYtOGY4OC0zNDM2M2JjN2M0YzAiOnsiZGVsZXRlIjo5MDAwMDAwMDAwMDB9fQ==",
          "dependencies": [
            "data.aws_vpc.vpc"
          ],
          "create_before_destroy": true
        }
      ]
    },
    {
      "module": "module.db.module.db_parameter_group",
      "mode": "managed",
      "type": "aws_db_parameter_group",
      "name": "this",
      "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
      "instances": [
        {
          "index_key": 0,
          "schema_version": 0,
          "attributes": {
            "arn": "arn:aws:rds:eu-west-2:xxx:pg:confluence-20210107151803555000000001",
            "description": "Database parameter group for confluence",
            "family": "postgres12",
            "id": "confluence-20210107151803555000000001",
            "name": "confluence-20210107151803555000000001",
            "name_prefix": "confluence-",
            "parameter": [],
            "tags": {
              "Application": "RDS",
              "Automated-stop-start": "true",
              "Environment": "xxx-vpc",
              "Name": "confluence",
              "Terraform": "true"
            }
          },
          "sensitive_attributes": [],
          "private": "bnVsbA==",
          "dependencies": [
            "data.aws_vpc.vpc"
          ],
          "create_before_destroy": true
        }
      ]
    },
    {
      "module": "module.db.module.db_subnet_group",
      "mode": "managed",
      "type": "aws_db_subnet_group",
      "name": "this",
      "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
      "instances": [
        {
          "index_key": 0,
          "schema_version": 0,
          "attributes": {
            "arn": "arn:aws:rds:eu-west-2:xxx:subgrp:confluence-20210107152452716900000003",
            "description": "Database subnet group for confluence",
            "id": "confluence-20210107152452716900000003",
            "name": "confluence-20210107152452716900000003",
            "name_prefix": "confluence-",
            "subnet_ids": [
              "subnet-03b546fe74d264112",
              "subnet-0422b9ca26db2e6e6",
              "subnet-0d596045ea93e149d"
            ],
            "tags": {
              "Application": "RDS",
              "Automated-stop-start": "true",
              "Environment": "xxx-vpc",
              "Name": "confluence",
              "Terraform": "true"
            }
          },
          "sensitive_attributes": [],
          "private": "bnVsbA==",
          "dependencies": [
            "data.aws_subnet_ids.subnets",
            "data.aws_vpc.vpc"
          ]
        }
      ]
    }
```

**Comment by eliecharra** [comment]
Thanks a lot for this detailed report @Staggerlee011 it is very appreciated 🙏🏻 

I will investigate this one ASAP

**Comment by eliecharra** [comment]
Will be fixed in the next 0.7.0 version

**Comment by sjourdan** [comment]
LGTM I confirm all of those disappeared with 0.7.0 in my case.
